students/ericstreit/lessons/lesson05/assignment/database.py

```diff
@@ -0,0 +1,217 @@
+                        logging.info(f'Adding {convert_to_dict} to the customers collection')
+                        record_count += 1
+                    except Exception as e:
+                        logging.error(f'Error adding {convert_to_dict} to the customers collection!')
+                        error_count += 1
+                for row in rentals_contents:
+                    logging.info(f'IMPORTING RENTALS DATA')
+                    convert_to_dict = {'product_id': row[0],
+                                       'user_id': row[1]}
+                    try:
+                        result = rentals.insert_one(convert_to_dict)
+                        logging.info(f'Adding {convert_to_dict} to the rentals collection')
+                        record_count += 1
+                    except Exception as e:
+                        logging.error(f'Error adding {convert_to_dict} to the rentals collection!')
+                        error_count += 1
+        except Exception as e:
+            logging.error(f'Hm, we had an error here?')
+            logging.error(e)
+
+    print(f' the number of records in the products collection is: {db.products.count()}')
+    print(f' the number of records in the customers collection is: {db.customers.count()}')
+    print(f' the number of records in the rentals collection is: {db.rentals.count()}')
+    return (record_count, error_count)
+
+def show_available_products():
+    """
+    Returns a Python dictionary of products listed as available with
+    the following fields:
+
+    product_id.
+    description.
+    product_type.
+    quantity_available.
+    """
+    mongo = MongoDBConnection()
+
+    with mongo:
+        #connect to Mongo
+        db = mongo.connection.hp_norton
+
+        #name the first collection we will work on
+        products = db["products"]
+        customers = db["customers"]
+        rentals = db["rentals"]
+
+        for name in products.find():
+            logging.debug(f"{name.get('product_id')} : {name.get('quantity_available')}")
+            try:
+                if int(name.get('quantity_available')) > 0:
+                    print(f"{name.get('product_id')}: {name.get('description')} is available!")
+            except ValueError as e:
+                logging.debug(f"skipping {name.get('product_id')} due to error")
+                logging.error(e)
+
+
+def show_rentals(product_id):
+    """
+    Returns a Python dictionary with the following user information from
+    users that have rented products matching product_id:
+
+    user_id.
+    name.
+    address.
+    phone_number.
+    email.
+
+    :param arg1: The product_id to search for
+    """
+    # gosh, I'm repeating myself a lot here with these Mongo commands. could
+    # be simplified into its own function, yes??
+    mongo = MongoDBConnection()
+
+    with mongo:
+        #connect to Mongo
+        db = mongo.connection.hp_norton
```

Comment:
Yes, you should not be connecting and reconnecting every time. There should be one connection when a user starts interacting with the database, and it should just stay open until they quit.